cleanlab/segmentation/filter.py

```diff
@@ -0,0 +1,180 @@
+        if len(labels.shape) != 3:
+            raise ValueError("labels must have a shape of (N, H, W)")
+
+        if len(pred_probs.shape) != 4:
+            raise ValueError("pred_probs must have a shape of (N, K, H, W)")
+
+        num_images, height, width = labels.shape
+        num_images_pred, num_classes, height_pred, width_pred = pred_probs.shape
+
+        if num_images != num_images_pred or height != height_pred or width != width_pred:
+            raise ValueError("labels and pred_probs must have matching dimensions for N, H, and W")
+        
+        #Check downsample
+        if height%downsample!=0 or width%downsample!=0:
+            raise ValueError(f"Height {height} and width {width} not divisible by downsample value of {downsample}")
+
+    
+    def downsample_arrays(labels: np.ndarray, pred_probs: np.ndarray, factor: int = 1) -> tuple[np.ndarray, np.ndarray]:
+        if factor == 1:
+            return labels, pred_probs
+        num_image, num_classes, h, w = pred_probs.shape
+        small_labels = np.round(labels.reshape((num_image, h // factor, factor,
+                                                w // factor, factor)).mean(4).mean(2))
+        small_pred_probs = pred_probs.reshape((num_image, num_classes, h // factor, factor,
+                                               w // factor, factor)).mean(5).mean(3)
+
+        # We want to make sure that pred_probs are renormalized
+        row_sums = small_pred_probs.sum(axis=1)
+        renorm_small_pred_probs = (small_pred_probs / np.expand_dims(row_sums, 1))
+
+        return small_labels, renorm_small_pred_probs
+    
+    def flatten_and_preprocess_masks(labels: np.ndarray, pred_probs: np.ndarray) -> tuple[np.ndarray, np.ndarray]:
+        _, num_classes, _, _ = pred_probs.shape
+        labels_flat = labels.flatten().astype(int)
+        pred_probs_flat = np.moveaxis(pred_probs, 0, 1).reshape(num_classes, -1)
+
+        return labels_flat, pred_probs_flat.T
+    
+    
+    ##
+    check_input(labels, pred_probs)
+    
+    
+    #Added Downsampling
+    pre_labels, pre_pred_probs = downsample_arrays(labels, pred_probs, downsample)
+    
+    num_image, num_classes, h, w = pre_pred_probs.shape
+    #flatten images just preps labels and pred_probs
+    
+    pre_labels, pre_pred_probs = flatten_and_preprocess_masks(pre_labels, pre_pred_probs)
+    
+    ranked_label_issues = find_label_issues_batched(pre_labels, pre_pred_probs, batch_size=batch_size, n_jobs=n_jobs,verbose=verbose)
+
+    
+    #Finding the right indicies
+    relative_index = ranked_label_issues % (h*w)
+    pixel_coor_i,pixel_coor_j  = np.unravel_index(relative_index, (h,w))
+    image_num = ranked_label_issues//(h*w)
+    
+    if scores_only:
+        return 1 - np.bincount(image_num)/(h*w)
+
+    #Upsample carefully maintaining indicies
+    img = np.full((num_image, h, w), False)
+
+    for num,ii,jj in zip(image_num,pixel_coor_i, pixel_coor_j):
```

Comment:
You use `img` and `image_num`, I would try and avoid shortening words and `img`->`image`